Source/Core/VideoCommon/HiresTextures.cpp

```diff
@@ -135,25 +145,74 @@ void HiresTexture::Update()
+    s_textureCacheAbortPrefetching.Clear();
     s_prefetcher = std::thread(Prefetch);
   }
 }
 
 void HiresTexture::Prefetch()
 {
+  // prefetch() is running in a tight loop, and it's using that to allow "at least" 1 Search()
+  // run per cycle
   Common::SetCurrentThreadName("Prefetcher");
 
-  size_t size_sum = 0;
-  size_t sys_mem = Common::MemPhysical();
-  size_t recommended_min_mem = 2 * size_t(1024 * 1024 * 1024);
-  // keep 2GB memory for system stability if system RAM is 4GB+ - use half of memory in other cases
-  size_t max_mem =
-      (sys_mem / 2 < recommended_min_mem) ? (sys_mem / 2) : (sys_mem - recommended_min_mem);
-  u32 starttime = Common::Timer::GetTimeMs();
-  for (const auto& entry : s_textureMap)
+  double prefetch_start_time_total = 0;
+  double prefetch_stop_time_total = 0;
+  constexpr double CULLED_PERCENT_DEFAULT = 30.0;
+  constexpr double CULLED_PERCENT_LOW_BOUND = 5.0;
+  constexpr double CULLED_PERCENT_HIGH_BOUND = 90.0;
+
+  // Overridable by HiresTexOverridePrefetchCulledPercent INI setting
+  double culled_percent = CULLED_PERCENT_DEFAULT;
+
+  if (Config::Get(Config::GFX_HIRESTEX_OVERRIDE_PREFETCH_CULLED_PERCENT) != CULLED_PERCENT_DEFAULT)
   {
-    const std::string& base_filename = entry.first;
+    if (Config::Get(Config::GFX_HIRESTEX_OVERRIDE_PREFETCH_CULLED_PERCENT) <
+        CULLED_PERCENT_LOW_BOUND)
+    {
+      culled_percent = CULLED_PERCENT_LOW_BOUND;
+      WARN_LOG(VIDEO, "Prefetcher: Mem limit cull percentage exceeds the minimum of %.2f %%. "
+                      "Raising the percentage to the lower bound.",
+               CULLED_PERCENT_LOW_BOUND);
+    }
+    else if (Config::Get(Config::GFX_HIRESTEX_OVERRIDE_PREFETCH_CULLED_PERCENT) >
+             CULLED_PERCENT_HIGH_BOUND)
+    {
+      culled_percent = CULLED_PERCENT_HIGH_BOUND;
+      WARN_LOG(VIDEO, "Prefetcher: Mem limit cull percentage exceeds the maximum of %.2f %%. "
+                      "Lowering the percentage to the upper bound.",
+               CULLED_PERCENT_HIGH_BOUND);
+    }
+    else if ((Config::Get(Config::GFX_HIRESTEX_OVERRIDE_PREFETCH_CULLED_PERCENT) >=
+              CULLED_PERCENT_LOW_BOUND) ||
+             (Config::Get(Config::GFX_HIRESTEX_OVERRIDE_PREFETCH_CULLED_PERCENT) <=
+              CULLED_PERCENT_HIGH_BOUND))
+    {
+      culled_percent = Config::Get(Config::GFX_HIRESTEX_OVERRIDE_PREFETCH_CULLED_PERCENT);
+      WARN_LOG(VIDEO, "Prefetcher: Mem limit cull percentage successfully overriden by user INI, "
+                      "at %.2f %%.",
+               culled_percent);
+    }
+  }
+
+  double prefetched_size_sum = 0;  // total mem size incl. non-native decompressed textures
+  const double absolute_max_mem = Common::MemPhysical();  // excl. HW reserved mem
+  const double culled_amount = (Common::MemPhysical() * culled_percent) / 100;
+  const double delegated_max_mem = Common::MemPhysical() - culled_amount;
+
+  if (absolute_max_mem <= DOLPHIN_MIN_REQ_MEM)
```

Comment:
IMO this check isn't really needed, it's safe to assume that systems have >=2GB RAM nowdays. We don't check it anywhere else, and if we did, it would better off in the general startup code, not the prefetcher.

Reply:
I'll wait a bit for anyone else to weight in here.